src/app/games/lasca/lasca.component.ts

```diff
@@ -0,0 +1,330 @@
+    private lastMove: MGPOptional<LascaMove> = MGPOptional.empty();
+    private validClicks: Coord[] = [];
+    private captureds: Coord[] = [];
+    private validCaptures: LascaMove[] = [];
+    private validSteps: LascaMove[] = [];
+    public LascaSpace: typeof LascaSpace = LascaSpace;
+
+    public constructor(messageDisplayer: MessageDisplayer) {
+        super(messageDisplayer);
+        this.hasAsymetricBoard = true;
+        this.rules = new LascaRules(LascaState);
+        this.availableMinimaxes = [
+            new LascaControlMinimax(this.rules, 'Control Minimax'),
+            new LascaControlAndDominationMinimax(this.rules, 'Control and Domination'),
+        ];
+        this.encoder = LascaMove.encoder;
+        this.tutorial = new LascaTutorial().tutorial;
+        this.canPass = false;
+        this.updateBoard();
+    }
+    public backgroundColor(x: number, y: number): string {
+        if ((x + y) % 2 === 0) {
+            return 'lightgrey';
+        } else {
+            return 'black';
+        }
+    }
+    public updateBoard(): void {
+        this.lastMove = this.rules.node.move;
+        const state: LascaState = this.getState();
+        this.board = state.getCopiedBoard();
+        this.validCaptures = LascaRules.getCaptures(state);
+        this.validSteps = LascaRules.getSteps(state);
+        this.updateAdaptedBoardFrom(state);
+        this.showLastMove();
+        this.showPossiblesMoves();
+        this.rotateAdaptedBoard();
+    }
+    private updateAdaptedBoardFrom(state: LascaState): void {
+        this.adaptedBoard = [];
+        for (let y: number = 0; y < 7; y++) {
+            const newRow: SpaceInfo[] = [];
+            for (let x: number = 0; x < 7; x++) {
+                const newSpace: SpaceInfo = this.getSpaceInfo(state, x, y);
+                newRow.push(newSpace);
+            }
+            this.adaptedBoard.push(newRow);
+        }
+        const selectedPiece: MGPOptional<Coord> = MGPOptional.ofNullable(this.validClicks[this.validClicks.length - 1]);
+        if (selectedPiece.isPresent()) {
+            const coord: Coord = selectedPiece.get();
+            for (const pieceInfo of this.adaptedBoard[coord.y][coord.x].pieceInfos) {
+                pieceInfo.classes.push('selected-stroke');
+            }
+            this.showPossibleLandings(coord, state);
+        }
+        for (const captured of this.captureds) {
+            this.adaptedBoard[captured.y][captured.x].spaceClasses.push('captured-fill');
+        }
+        for (const moved of this.validClicks) {
+            this.adaptedBoard[moved.y][moved.x].spaceClasses.push('moved-fill');
+        }
+    }
+    private getSpaceInfo(state: LascaState, x: number, y: number): SpaceInfo {
+        const space: LascaSpace = state.getPieceAtXY(x, y);
+        const pieceInfos: LascaPieceInfo[] = [];
+        let pieceIndex: number = space.getPileSize() - 1; // Start by the lower piece
+        while (pieceIndex >= 0) {
```

Comment:
for (let pieceIndex = ...; pieceIndex >= 0; pieceIndex--) ?

for est plus lisible que while pour des simples comptages !